test/fail_compilation/test16228.d

```diff
@@ -0,0 +1,24 @@
+/* REQUIRED_ARGS: -dip25
+   TEST_OUTPUT:
+---
+fail_compilation/test16228.d(22): Error: function has 'return' but does not return any indirections
```

Comment:
`Error: function has 'return' but does not return any instance variable (or part of) by reference` sounds better to me.